server/models/Notification.js

```diff
@@ -0,0 +1,84 @@
+const mongoose = require("mongoose");
+
+const notifyTypes = ["user", "message"];
+const requireSender = ["user", "message"];
+
+const notificationSchema = new mongoose.Schema(
+  {
+    notifyType: {
+      type: String,
+      enum: notifyTypes,
+      required: [
+        true,
+        `A type of notification is required. Options: ${notifyTypes.join(
+          ", "
+        )}`,
+      ],
+    },
+    title: {
+      type: String,
+      required: [true, "A notification title is required."],
+      trim: true,
+      maxlength: 40,
+    },
+    description: {
+      type: String,
+      required: [true, "A description of the notification is required."],
+      trim: true,
+      maxlength: 150,
+    },
+    readBy: {
+      type: [
+        {
+          receiverId: {
+            type: mongoose.Types.ObjectId,
+            ref: "User",
+          },
+          readAt: { type: Date, default: Date.now() },
+        },
+      ],
+      default: [],
+      _id: false,
+    },
+    sender: {
+      type: mongoose.Types.ObjectId,
+      ref: "Profile",
+      required: [
+        requiresSender,
+        "Notification type requires a user as a sender.",
+      ],
+    },
+    receivers: {
+      type: [
+        {
+          type: mongoose.Types.ObjectId,
+          ref: "User",
+        },
+      ],
+      validate: [
+        validReceivers,
+        "Must contain at least one recipient and/or sender cannot be included within receivers.",
+      ],
+    },
```

Comment:
This only occured to me while I was going through the controllers but having multiple recipients may have been the wrong way to go about this. It would have been much simpler to have a single recipient for each notification. If an event requires multiple users to be notified, we can simply create multiple notifications, one for each recipient. I'm okay with keeping things as they are in order to throw away all your efforts in making this work but I thought this was worth mentioning still 👍🏼

Reply:
The reason I chose this method over just having one sender and receiver is for the case we have notifications that reach beyond peer to peer. In which case, overloading both the route and the database with several notifications, would not be ideal.